src/cloud/apis/reporting.ts

```diff
@@ -51,3 +56,28 @@ export const reportPoints = (batch: PointsBatch) => {
     }
   }
 }
+
+const asRudderOptions = (tags: PointTags, campaign: object) => {
+  const options = {}
+  let context = null
+  Object.keys(tags).forEach((key: string) => {
+    const value = tags[key]
+    if (key != 'context') {
+      options[key] = value
+    } else {
+      if (typeof value != 'object') {
+        // context is reserved rudder element key and must be object, see rudder-sdk-js (#20463)
+        options['appContext'] = value
+      } else {
+        context = options['context'] = value
+      }
+    }
+  })
+  if (Object.keys(campaign).length) {
+    if (context == null) {
+      context = options['context'] = {}
```

Comment:
again unclear on what this should do

Reply:
I have refactored it, hopefully it is better now.